zerver/views/test_emails.py

```diff
@@ -59,24 +59,21 @@ def email_page(request):
         'organization_setup_advice_link': '%s/help/getting-your-organization-started-with-zulip' % (settings.ROOT_DOMAIN_URI,),
     }
 
-    templates = [
-        'confirm_registration', 'invitation', 'invitation_reminder', 'followup_day1',
-        'followup_day2', 'missed_message', 'digest', 'find_team', 'password_reset',
-        'confirm_new_email', 'notify_change_in_email', 'notify_new_login']
+    templates = set()
 
     for f in os.listdir(os.path.join(ZULIP_PATH, 'templates', 'zerver', 'emails')):
-        template = f.split('.')[0]
-        if template not in templates:
-            templates.append(template)
+        template = f.split('.source.html')[0]
+        if template != f:
+            templates.add(template)
```

Comment:
What's the story with this block of code?  I don't understand it, so we should either make a comment or clean it up some other way.

Reply:
If the file name does not have `.source.html` extension `f.split('.source.html')[0]` will return the file name itself. Since we are only interested in `.source.html` file types we skip them.

Reply:
can we just do `if f.endswith(".source.html")` ?  That reads much more clearly.

Reply:
Yeah. That sounds way better.